src/main.c

```diff
@@ -0,0 +1,305 @@
+/**
+ * @file main.c
+ * @author David Chung 
+ * @brief Main entry point for simple-calc.
+ * 
+ * This program performs simple arithmetic calculations given user input via CLI.
+ * 
+ * @version 3.0
+ * @date 2024-11-25
+ *
+ */
+
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <stdbool.h>
+#include <errno.h>
+#include <string.h>
+#include "arithmetic.h"
+#include "debug.h"
+#include "helpers.h"
+
+typedef enum
+{
+    ADD,
+    SUBTRACT,
+    MULTIPLY,
+    DIVIDE,
+    INVALID
+} Operation;
+    
+Operation checkOperator(const char *operator)
+{
+    if (strcmp(operator, "+") == 0)
+    { 
+        return ADD;
+    }
+    else if (strcmp(operator, "-") == 0) 
+    {
+        return SUBTRACT;
+    }
+    else if (strcmp(operator, "x") == 0)
+    { 
+        return MULTIPLY;
+    }
+    else if (strcmp(operator, "/") == 0)
+    {
+        return DIVIDE;
+    }
+    else 
+        return INVALID;
+}
+
+/**
+ * @brief Main function for the simple-calc program
+ * 
+ * @param argc Number of CMD Line arguments
+ * @param argv Array of CMD Line argument strings 
+ * @return int 0 if successful.
+ */
+
+int main(int argc, char *argv[]) {
+
+    Operation operation;
+
+    //int opt = 0;
+    long num1, num2;                /**< Numeric arguments will be converted to a long int. */
```

Comment:
Initialize everything

Numeric values to 0 or -1 as appropriate

Structs to {0}

Pointers to NULL